### PR TITLE
Simplify tqueue

### PR DESCRIPTION
**Pull request by alagner**
Some code clean-up, mostly moving from bare loops to STL algos/ranged for loops.
I actually wanted to use upper bound in `activate0`, too, but it seems objects are deletign themselves sometimes, which ends in double-free again, so I gave it up, temporarily at least.
@marzojr two questions related to tqueue:
1. What will you say about using a different data structure? As it's supposed to be sorted, multimap with time being the key seems more viable than list. I was also thinking about priority queue, but it does not allow random access.
2. What do you think about including Boost? Some classes like Tqeue could greatly benefit from it, e.g. thanks to its filter_iterator (which currently Tqeue_iteratator replaces). License-wise that should not be a problem, technically not a big issue either, as IIRC when iterator is a header-only lib, question is if an extra dependency is to be tolerated there.

### CODE REVIEW

File: tqueue.cc [hunk 1]
```diff
@@ -82,15 +70,14 @@ bool    operator <(const Queue_entry &q1, const Queue_entry &q2) {
...
-			obj->queue_cnt--;
-			data.erase(it);
-			return 1;
-		}
+	} else {
+		(*toRemove).handler->queue_cnt--;
+		data.erase(toRemove);
+		return 1;
 	}
 	return 0;         // Not found.
```
Comment (marzojr) [inline tqueue.cc:82]: This is dead code.

File: tqueue.cc [hunk 2]
```diff
@@ -82,15 +70,14 @@ bool    operator <(const Queue_entry &q1, const Queue_entry &q2) {
...
-		if (it->handler == obj) {
-			obj->queue_cnt--;
-			data.erase(it);
-			return 1;
-		}
+	} else {
+		(*toRemove).handler->queue_cnt--;
+		data.erase(toRemove);
+		return 1;
 	}
```
Comment (marzojr) [inline tqueue.cc:81]: Please avoid using `else` here, as the `if` is just a return. Maybe invert the condition check? See also dead code comment.

File: tqueue.cc [hunk 3]
```diff
@@ -105,17 +92,15 @@ int Time_queue::remove(
...
+    auto it = std::find_if(data.begin(), data.end(),
+            [&](const auto& el) { return el.handler == obj && el.udata == udata; }
+    );
+    bool found = it != data.end();
+    if (found) {
+		obj->queue_cnt--;
+		data.erase(it);
 	}
-	return 0;         // Not found.
+    return found;
```
Comment (marzojr) [inline tqueue.cc:103]: Indenting with spaces instead of tabs.

**Review by marzojr — CHANGES_REQUESTED** [review]

**Review by marzojr — APPROVED** [review]

**Comment by DominusExult** [comment]
This is causing regressions in #143 and #140

**Pull request merged**